src/main/java/tech/ydb/io/r2dbc/YdbConnection.java

```diff
@@ -23,24 +23,32 @@
 import io.r2dbc.spi.Statement;
 import io.r2dbc.spi.TransactionDefinition;
 import io.r2dbc.spi.ValidationDepth;
+
 import java.time.Duration;
+
 import reactor.core.publisher.Mono;
-import tech.ydb.io.r2dbc.state.YDBConnectionState;
-import tech.ydb.table.TableClient;
+import tech.ydb.io.r2dbc.query.R2dbcQueryParser;
+import tech.ydb.io.r2dbc.query.YdbQuery;
+import tech.ydb.io.r2dbc.state.YdbConnectionState;
+import tech.ydb.io.r2dbc.statement.DataStatement;
+import tech.ydb.io.r2dbc.statement.SchemaStatement;
+import tech.ydb.io.r2dbc.type.YdbTypeResolver;
 
 /**
  * @author Kirill Kurdyukov
  */
-public class YDBConnection implements Connection {
-
-    private final TableClient tableClient;
+public class YdbConnection implements Connection {
+    private final YdbTypeResolver typeResolver;
+    private final R2dbcQueryParser r2dbcQueryParser;
 
     private volatile IsolationLevel isolationLevel = IsolationLevel.SERIALIZABLE;
     private volatile boolean autoCommit = true;
-    private volatile YDBConnectionState state;
+    private final YdbConnectionState state;
```

Comment:
Why is state final? I have exec out transaction query, then start transaction.

Reply:
Revert to volatile.